src/main/java/ui/UI.java

```diff
@@ -67,8 +67,7 @@ public void sayHi() {
      * Prints a farewell message to the console.
      */
     public void cleanupAndExit() {
-        printResponse("Adios, amigo!");
-        scanner.close();
```

Comment:
The `scanner.close()` should be called to prevent resource leaks. Perhaps you forgot to add it back after refactoring the exit function 😅